Documentation/ceph-cluster-crd.md

```diff
@@ -815,12 +815,25 @@ So just taking the `storage` section this will give something like:
           volumeMode: Block
           accessModes:
             - ReadWriteOnce
+      - metadata:
+          name: waldata
```

Comment:
If this option is specified, is the `metadata` template required? Or what happens if only `data` and `wal` are specified?

Reply:
It is a good question, actually the ceph-volume for bluetore support the following cases:
1. A block device, a block.wal, and a block.db device
2. A block device and a block.wal device
3. A block device and a block.db device
4. A single block device
so I will rework the code to support all the  cases.
It will be a little complicated, when you assign 2 devices, we have 2 conditions, block + block.db and block + block.wal.

Reply:
I am working on the re-construction to support the 4 options for block, db and wal.  But there is a problem:

In pkg/daemon/ceph/osd/daemon.go Provision() function, I have to add rawDevices by there bluestore partition types: data, metadata or wal, currently the implementation is depends on the index and length of agent.devices (index 0 for block, 1 for metadata, 2 for wal), if I rework to based on the actual configurations in volumeClaimTemplates yaml, I need to get the osds.DesiredStorage.VolumeSources, but it is hard to get the hander of osd manager (osds) from daemon

Do you have any suggestion for it?

Reply:
@silverhandy There was a recent change that will hopefully simplify your implementation of this feature. With #5825, the osd settings are passed through json serialization instead of through raw index parsing. Please rebase on that change and see if that helps answer the question. There are a number of merge conflicts now for this PR, so feel free to ping us in the #ceph-dev channel in the Rook Slack for help with that.

Reply:
I have updated the PR and verified all the 4 cases as following, it is depend by device type not index.
1/ A block device, a block.wal, and a block.db device
2/ A block device and a block.wal device
3/ A block device and a block.db device
4/ A single block device